src/Solvers/conjugate_gradient_poisson_solver.jl

```diff
@@ -84,8 +84,8 @@ struct DefaultPreconditioner end
 """
     ConjugateGradientPoissonSolver(grid;
                                    preconditioner = DefaultPreconditioner(),
-                                   reltol = sqrt(eps(grid)),
-                                   abstol = sqrt(eps(grid)),
+                                   reltol = min(100 * eps(grid), 100 * eps(grid) * minimum_cell_volume(grid)^2, sqrt(eps(grid))),
+                                   abstol = min(100 * eps(grid), sqrt(eps(grid))),
```

Comment:
can we fix this so its more readable? something like `reltol = default_rel_tol`. And then describe the `reltol` later on in the docstring? Same for abstol.